flight-server/src/main/java/org/m4d/adp/flight_server/RelayProducer.java

```diff
@@ -69,6 +69,7 @@ public RelayProducer(Location location, Location remote_location,
         transformed_buffer.get(transformedRecordBatchByteArray);
         // Deallocate transformed bytes
         AllocatorInterface.wasmDealloc(instance_ptr, transformed_bytes_address, transformed_bytes_len);
```

Comment:
You didn't allocate it explicitly so it's weird to me to see it released here.
In my version I changed the code so these would be deallocated internally by drop_tuple instead:
```rust
#[no_mangle]
pub fn drop_tuple(tuple_ptr: i64) {
    let tuple = Into::<Pointer<Tuple>>::into(tuple_ptr);
    let tuple = tuple.deref();
    unsafe {
        drop(Vec::from_raw_parts(tuple.0 as *mut u8, tuple.1 as usize, tuple.1 as usize));
    };
}
```